src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -2631,6 +2631,16 @@ public void ImportLibrary(object parameter)
                 {
                     System.Windows.MessageBox.Show(String.Format(ex.Message, MessageBoxButtons.OK, MessageBoxIcon.Warning));
                 }
+                catch(FileLoadException ex)
+                {
+                    // If the exception is having HRESULT of 0x80131515, then we need to instruct the user to "unblock" the downloaded DLL.
+                    if (ex.HResult == unchecked((int)0x80131515))
+                    {
+                        var failureMessage = string.Format(Properties.Resources.LibraryLoadFailureForBlockedAssembly, ex.Message);
+                        System.Windows.MessageBox.Show(failureMessage, BrandingResourceProvider.ProductName, MessageBoxButton.OK, MessageBoxImage.Warning);
+                    }
```

Comment:
All such exceptions that are thrown during loading of blocked assemblies explicitly imported end up here. A message box is displayed to the user, and Dynamo continues to work normally.